### ISSUE TITLE
The method of calculating physical loss is incorrect

### ISSUE THREAD

**Issue by SKY-longlong** [viviRG2024/PINNGNN#3]
The author did not use automatic differentiation when generating physicals_loss.
For example, in code:
continuity_loss = torch.mean(torch.abs((rho[:, 1:, :] - rho[:, :-1, :]) / dt +  (q[:, :, 1:] - q[:, :, :-1]) / dx))
I think it should be:
continuity_loss = torch.mean(torch.abs(torch.autograd.grad(rho,t)+torch.autograd.grad(q,x)))

**Comment by alphonse-lin** [comment]
Thank you so much for bringing this. We would review the code and correct this
